bolt/lib/Passes/IndirectCallPromotion.cpp

```diff
@@ -1467,7 +1466,7 @@ void IndirectCallPromotion::runOnFunctions(BinaryContext &BC) {
                                std::max<uint64_t>(TotalIndexBasedCandidates, 1))
          << "%\n";
 
-  (void)verifyProfile;
+  (void)verifyProfile(BFs);
```

Comment:
This is not the same. This will run verifyProfile(), which will cause an unnecessary slow down in no-debug mode. The previous code wasn't running anything, it was just making a reference to silence a compiler warning about verifyProfile being unused.

I believe the proper fix is actually to surround the definition of verifyProfile in #ifndef NDEBUG..#endif in the same way as it is use is.

Reply:
Thank you for your review.  Do you mean I need to delete this line of code?

@rafaelauler

Reply:
Correct. Delete the line entirely and surround the definition of verifyProfile with #ifndef, like that:

#ifndef NDEBUG
definition goes here
#endif

Reply:
The code has been modified, please review it, thank you!

@rafaelauler